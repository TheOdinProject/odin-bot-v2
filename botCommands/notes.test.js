const command = require('./notes');

describe('?notes', () => {
  describe('regex', () => {
    it.each([
      ['?notes'],
      [' ?notes'],
      ['?notes @odin-bot'],
      ['@odin-bot ?notes'],
    ])('correct strings trigger the callback', (string) => {
      expect(command.regex.test(string)).toBeTruthy();
    });

    it.each([
      ['ntoes'],
      ['notes'],
      ['noets'],
      ['?noets'],
      ['?ntoes'],
      ['? notes'],
      ['?anotes'],
    ])("'%s' does not trigger the callback", (string) => {
      expect(command.regex.test(string)).toBeFalsy();
    });

    it.each([
      ['Check this out! ?notes'],
      ['Don\'t worry about ?notes'],
      ['Hey @odin-bot, ?notes'],
      ['?@odin-bot ^ ?me ?notes ?tests$*'],
    ])("'%s' - command can be anywhere in the string", (string) => {
      expect(command.regex.test(string)).toBeTruthy();
    });

    it.each([
      ['@user?notes'],
      ['it\'s about?notes'],
      ['?notesanillusion'],
      ['?notes?'],
      ['?notes*'],
      ['?notes...'],
    ])("'%s' - command should be its own word/group - no leading or trailing characters", (string) => {
      expect(command.regex.test(string)).toBeFalsy();
    });
  });

  describe('callback', () => {
    it('returns correct output', () => {
      expect(command.cb()).toMatchSnapshot();
    });
  });
});
