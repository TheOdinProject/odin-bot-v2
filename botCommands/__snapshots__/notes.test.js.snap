// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`?notes callback returns correct output 1`] = `"To note, or not to note, that is the question. Please read this: https://discord.com/channels/505093832157691914/505093832157691916/768161823366578176"`;
